From 21d41d5d8b1ea3d6005344b66d7911f8c437007d Mon Sep 17 00:00:00 2001
From: Colin Cross <ccross@android.com>
Date: Tue, 11 Mar 2014 18:19:53 -0700
Subject: [PATCH 023/634] HACK: input: evdev: disable EVIOCREVOKE

Android used the same ioctl signature as EVOICREVOKE for
EVIOCSSUSPENDBLOCK.  Disable EVIOCREVOKE until Android userspace
is available that uses EPOLLWAKEUP instead.

Change-Id: I51aa63fd19b20a3966bef1c46d1bb845c2589a5a
Signed-off-by: Colin Cross <ccross@android.com>
---
 drivers/input/evdev.c      | 13 ++++++++++++-
 include/uapi/linux/input.h |  4 +++-
 2 files changed, 15 insertions(+), 2 deletions(-)

diff --git a/drivers/input/evdev.c b/drivers/input/evdev.c
index bb9f073..73553f7 100644
--- a/drivers/input/evdev.c
+++ b/drivers/input/evdev.c
@@ -809,6 +809,11 @@ static int evdev_handle_mt_request(struct input_dev *dev,
 	return 0;
 }
 
+/*
+ * HACK: disable conflicting EVIOCREVOKE until Android userspace stops using
+ * EVIOCSSUSPENDBLOCK
+ */
+/*
 static int evdev_revoke(struct evdev *evdev, struct evdev_client *client,
 			struct file *file)
 {
@@ -819,6 +824,7 @@ static int evdev_revoke(struct evdev *evdev, struct evdev_client *client,
 
 	return 0;
 }
+*/
 
 static long evdev_do_ioctl(struct file *file, unsigned int cmd,
 			   void __user *p, int compat_mode)
@@ -882,12 +888,17 @@ static long evdev_do_ioctl(struct file *file, unsigned int cmd,
 		else
 			return evdev_ungrab(evdev, client);
 
+	/*
+	 * HACK: disable conflicting EVIOCREVOKE until Android userspace stops
+	 * using EVIOCSSUSPENDBLOCK
+	 */
+	/*
 	case EVIOCREVOKE:
 		if (p)
 			return -EINVAL;
 		else
 			return evdev_revoke(evdev, client, file);
-
+	*/
 	case EVIOCSCLOCKID:
 		if (copy_from_user(&i, p, sizeof(unsigned int)))
 			return -EFAULT;
diff --git a/include/uapi/linux/input.h b/include/uapi/linux/input.h
index f484952..007cd35 100644
--- a/include/uapi/linux/input.h
+++ b/include/uapi/linux/input.h
@@ -152,7 +152,9 @@ struct input_keymap_entry {
 #define EVIOCGEFFECTS		_IOR('E', 0x84, int)			/* Report number of effects playable at the same time */
 
 #define EVIOCGRAB		_IOW('E', 0x90, int)			/* Grab/Release device */
-#define EVIOCREVOKE		_IOW('E', 0x91, int)			/* Revoke device access */
+
+/* HACK: disable conflicting EVIOCREVOKE until Android userspace stops using EVIOCSSUSPENDBLOCK */
+/*#define EVIOCREVOKE		_IOW('E', 0x91, int)*/			/* Revoke device access */
 
 #define EVIOCSCLOCKID		_IOW('E', 0xa0, int)			/* Set clockid to be used for timestamps */
 
-- 
1.9.1

